crates/ruff_python_parser/src/lexer.rs

```diff
@@ -580,18 +580,53 @@ impl<'source> Lexer<'source> {
             }
         }
 
-        let mut indentation = Indentation::root();
+        if self.state.is_after_newline() {
+            if let Some(indentation) = self.eat_indentation()? {
+                return Ok(indentation);
+            }
+        } else {
+            self.skip_whitespace()?;
+        }
+
```

Comment:
If I understand the diff correctly, the whitespace lexing loop is repeated twice. If we're right after the newline only then we'll try to handle indentation otherwise we'll use the fast loop to _not_ handle indentation and skip any whitespace.

Reply:
Yes, that's correct